lang/cs/Org.Apache.REEF.Bridge.Core.Client/Grpc/Runtime/YarnRuntimeParametersProvider.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+using System.Collections.Generic;
+using Org.Apache.REEF.Bridge.Core.Client.Grpc.Config.Runtime.Parameters.Yarn;
+using Org.Apache.REEF.Bridge.Proto;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Bridge.Core.Client.Grpc.Runtime
+{
+    internal sealed class YarnRuntimeParametersProvider : IRuntimeParametersProvider
+    {
+
+        private readonly YarnRuntimeParameters _yarnRuntimeParameters;
+
+        [Inject]
+        private YarnRuntimeParametersProvider(
+            [Parameter(Value = typeof(JobPriority))] int jobPriority,
+            [Parameter(Value = typeof(JobQueue))] string jobQueue,
+            [Parameter(Value = typeof(JobSubmissionDirectoryPrefix))] string jobSubmissionDirectoryPrefix,
+            [Parameter(Value = typeof(SecurityTokenStrings))] ISet<string> securityTokenStrings,
+            [Parameter(Value = typeof(UnmanagedDriver))] bool unmanagedDriver,
+            [Parameter(Value = typeof(FileSystemUrl))] string fileSystemUrl)
+        {
+            _yarnRuntimeParameters = new YarnRuntimeParameters()
+            {
+                Priority = (uint) jobPriority,
```

Comment:
This seems dangerous. Shouldn't this be checking for `<0` first.

Reply:
@singlis I totally agree, and is what I started off with until I executed and got:
```
System.NotSupportedException
Encountered unknown primitive type!
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Org.Apache.REEF.Tang.Util.ReflectionUtilities.BoxClass(Type c)
   at Org.Apache.REEF.Tang.Implementations.ClassHierarchy.ClassHierarchyImpl.RegisterType(Type type)
   at Org.Apache.REEF.Tang.Implementations.ClassHierarchy.ClassHierarchyImpl.GetNode(Type type)
```

Reply:
p.s., yes @markusweimer I'll file a bug with Tang :)